File: examples/java/Stack1.java
public class Stack1 {
    static int[] arr = new int[100];
    static int top = -1;

    static void push(int x) {
        if (top < arr.length - 1) arr[++top] = x;
    }

    static int pop() {
        return top >= 0 ? arr[top--] : -1;
    }

    static int peek() {
        return top >= 0 ? arr[top] : -1;
    }

    public static void main(String[] args) {
        push(10);
        push(20);
        push(30);
        System.out.println(pop());
        System.out.println(peek());
    }
}

File: examples/csharp/Stack1.cs
using System;

class Stack1
{
    static void Main()
    {
        int[] stack = new int[10];
        int top = -1;
        top = Push(stack, top, 5);
        top = Push(stack, top, 10);
        top = Push(stack, top, 15);
        Console.WriteLine(Pop(stack, ref top));
        Console.WriteLine(Pop(stack, ref top));
    }

    static int Push(int[] stack, int top, int val)
    {
        stack[++top] = val;
        return top;
    }

    static int Pop(int[] stack, ref int top)
    {
        return top >= 0 ? stack[top--] : -1;
    }
}
